ubg/src/mafia/MafiaNight.js

```diff
@@ -30,44 +30,101 @@ const useStyles = makeStyles((theme) => ({
 
   /**
    * @return {undefined}
    */
   function loadNightData() {
-    const allPlayers = [];
-    usersData.forEach(function(u) {
-      if (u.alive === true) {
-        allPlayers.push(u);
-      }
-      if (u.uid === user.uid) {
-        setUserInfo(u);
-        if ( u.role=== 1) {
-          setRoleText('Pretend to be clicking or thinking :)');
-        } else if (u.role === 2) {
-          setRoleText('Mafia, pick someone to kill.');
-        } else if (u.role === 3) {
-          setRoleText('Detective, who do you want to check tonight?');
-        } else if (u.role === 4) {
-          setRoleText('Doctor, who do you want to save tonight?');
+    if (mafiaKill['uid'] !== '' &&
+    doctorSave['uid'] !== '' &&
+    detectiveCheck['uid'] !== '') {
+      room.update({day: true});
+    }
+    if (initialize === false) {
+      setInitialize(true);
+      const roles = new Set();
+      const allPlayers = [];
+      usersData.forEach(function(u) {
+        if (u.alive === true) {
+          roles.add(u.role);
+          allPlayers.push(u);
+        }
+        if (u.uid === user.uid) {
+          setUserInfo(u);
+          if ( u.role=== 1) {
+            setRoleText('Pretend to be clicking, tapping or thinking :)');
+          } else if (u.role === 2) {
+            setRoleText('Mafia, pick someone to kill.');
+          } else if (u.role === 3) {
+            setRoleText('Detective, who do you want to check tonight?');
+          } else if (u.role === 4) {
+            setRoleText('Doctor, who do you want to save tonight?');
+          }
+        }
+      });
+      // This is probably not best practice, tell me if you have better idea
+      // I think this should probably be done in Room.js?
+      for (let i = 2; i < 5; i++) {
+        if (!roles.has(i)) {
+          if (i === 2) {
+            room.update({mafiaKill: {uid: '#', displayName: ''}});
+          } else if (i === 3) {
+            room.update({detectiveCheck: {uid: '#', displayName: ''}});
+          } else {
+            room.update({doctorSave: {uid: '#', displayName: ''}});
+          }
         }
       }
-    });
-    setPlayers(allPlayers);
+      setPlayers(allPlayers);
+    }
   }
   /**
    * Load the all the mafia related data
    */
-  useEffect(loadNightData, []);
+  useEffect(loadNightData, [mafiaKill, doctorSave, detectiveCheck]);
   /**
    * @param {object} player information of player
    * @return {undefined}
    */
   function handleClick(player) {
-    console.log('You choose ' + player.displayName);
+    if (userInfo.role === 2) {
+      if (mafiaKill['uid'] === '') {
+        room.update(
+            {mafiaKill: {uid: player.uid, displayName: player.displayName}});
+        setMessage('You have killed ' + player.displayName + ' tonight.');
+      } else if (room.mafiaKill !== player.uid) {
+        setMessage('You have already chosen ' +
+        mafiaKill['displayName'] + ' to kill tonight.');
+      }
+    }
+    if (userInfo.role === 3) {
```

Comment:
I think you missed an else here, but maybe also consider a switch for this if else sequence.